### ISSUE TITLE
Aseprite is required to run the app on a programmers machine

### ISSUE THREAD

**Issue by tavurth** [viniciusgerevini/godot-aseprite-wizard#42]
When working with a friend I came across these errors while running the game. Aseprite is not installed on her machine.

```log
ERROR - Could not import aseprite file: Aseprite command failed. Please, check if the right command is in your PATH or configured through the "configuration" button.
 editor/editor_file_system.cpp:1755 - Error importing 'res://player/animations/SantaFrames.aseprite'.
ERROR - Could not import aseprite file: Aseprite command failed. Please, check if the right command is in your PATH or configured through the "configuration" button.
 editor/editor_file_system.cpp:1755 - Error importing 'res://worlds/city/lights/lantern.aseprite'.
ERROR - Could not import aseprite file: Aseprite command failed. Please, check if the right command is in your PATH or configured through the "configuration" button.
 editor/editor_file_system.cpp:1755 - Error importing 'res://worlds/city/lights/light01.aseprite'.
ERROR - Could not import aseprite file: Aseprite command failed. Please, check if the right command is in your PATH or configured through the "configuration" button.
 editor/editor_file_system.cpp:1755 - Error importing 'res://worlds/city/smoke/smoke.aseprite'.
```

Does this mean that Aseprite should be installed on all machines which intend to run this?

How does this work for exported games?

**Comment by viniciusgerevini** [comment]
Hello @tavurth,

I'm assuming you are using the automatic importer.

For development, if you do not include the `.import` folder in your version control (or when copying to the new machine) you need to have Aseprite installed. Even if you are just running the game, because it will try to re-import all the files on startup.

For exported games, Aseprite is not required, as you will be dealing with resource files already generated.

**Comment by tavurth** [comment]
Hi @viniciusgerevini, as far as the [Godot Docs](https://docs.godotengine.org/en/3.0/getting_started/workflow/assets/import_process.html#files-generated) state, adding this folder is optional, and it's always been something I've avoided adding.

So for best practices with this plugin a developer should include this folder? Perhaps we can add that to the docs?

**Comment by tavurth** [comment]
Hi @viniciusgerevini 

When importing on a second machine all of the imported sprites (from `*.aseprite`) are deleted automatically by Godot 3.4.2, as can be seen in this screenshot:

![telegram-cloud-photo-size-2-5350836846709555355-y](https://user-images.githubusercontent.com/100964/147390567-7ce35acd-9d82-452b-87bf-2b3d24b993c7.jpg)

So unfortunately committing the `.import` directory doesn't seem like a good solution at the moment.

**Comment by viniciusgerevini** [comment]
> So for best practices with this plugin a developer should include this folder? Perhaps we can add that to the docs?
Not really,

**Comment by viniciusgerevini** [comment]
> So for best practices with this plugin a developer should include this folder? 

Not really. This is the same trade off mentioned in Godot's documentation. If you don't include the folder, it needs to be re-imported in the new machine. As the plugin requires Aseprite, it's natural it will fail in case it's not present.

> When importing on a second machine all of the imported sprites (from *.aseprite) are deleted automatically by Godot 3.4.2, as can be seen in this screenshot:

This is weird. I haven't seen any changelog mentioning this. Let me try here.

**Comment by viniciusgerevini** [comment]
@tavurth Are you sure Godot is the one removing these files? I tested this by moving to different machines (linux, windows, macos) and it did not delete the files. I did see some changes related to file permission, but those didn't trigger a re-import.

Anyway, going back to the re-import issue. Unfortunately, by using the automatic importer Aseprite is added as a dependency to the devloop. Committing the import folder could be used as a workaround, but as you mentioned, it's not ideal in many cases, as it adds to the project size, and it seems to have some caveats in your case.

For this plugin, the only way I could remove the dependency would be adding aseprite executables as part of the project, what would actually bring a bunch of other problems and it wouldn't worth the effort.
